oidc/Parameters.mli

```diff
@@ -45,17 +45,18 @@ val make :
   ?display:display ->
   ?prompt:prompt ->
   ?nonce:string ->
-  Client.t ->
   redirect_uri:Uri.t ->
+  client_id:string ->
+  unit ->
   t
 
 val to_query : t -> (string * string list) list
 (** Used when starting a authentication *)
 
 val to_yojson : t -> Yojson.Safe.t
-val of_yojson : clients:Client.t list -> Yojson.Safe.t -> (t, [> error]) result
+val of_yojson : Yojson.Safe.t -> (t, [> error]) result
 
 (** {2 Parsing in the provider} *)
 
-val parse_query : clients:Client.t list -> Uri.t -> (t, [> error]) result
- 
\ No newline at end of file
+val parse_query : allowed_redirect_uris:string list -> Uri.t -> (t, [> error]) result
```

Comment:
A problem with this is that we probably don't know the redirect_uris unless we already know what client the request is coming to. We could maybe still pass in the list of clients but not put it in the record?

Reply:
I actually removed this altogether: https://github.com/ulrikstrid/ocaml-oidc/pull/29/commits/7b8f3e8768cbafc824588ccf67ad76bd11e6124e

We can't just pass the list of clients, because it could be unbounded if you have millions of clients in a server.

Reply:
I think a good alternative here is to provide a validation function after you have a `Parameters.t`, where you can provide a (just one) client (after you've looked it up based on `params.client_id`), and we validate it against the expected `redirect_uri`s

Reply:
Yepp, that's better 👌